app/api/submission/portfolio_api.rb

```diff
@@ -74,11 +77,13 @@ class PortfolioApi < Grape::API
 
         evidence_loc = project.portfolio_path
 
-        if evidence_loc.nil? || File.exist?(evidence_loc) == false
-          evidence_loc = Rails.root.join('public', 'resources', 'FileNotFound.pdf')
-          header['Content-Disposition'] = 'attachment; filename=FileNotFound.pdf'
-        else
-          header['Content-Disposition'] = 'attachment; filename=portfolio.pdf'
+        if params[:as_attachment]
+          if evidence_loc.nil? || File.exist?(evidence_loc) == false
+            evidence_loc = Rails.root.join('public', 'resources', 'FileNotFound.pdf')
+            header['Content-Disposition'] = 'attachment; filename=FileNotFound.pdf'
```

Comment:
I suggest we add a `filename` variable - set outside the attachment test.
Then within the attachment test if we can set the content disposition to use the filename from the variable. Then there is no need for the inner if here...